### PR TITLE
(PDK-589) Initial Readme entry for pdk convert

### PR DESCRIPTION
**Pull request by HelenCampbell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,44 @@ pdk test unit
 
 This command runs all available unit tests.
 
+### Convert an existing module
+
+Aside from creating new modules, the PDK now has the ability to convert an existing module into a PDK ?enabled? module. You can then manage and test your module using the tools bundled with the PDK.
+
+To convert a module you must first decide on a template to convert to. This as default is set as ?PUPPET TEMPLATE?, however you can use your own template by passing it in as a parameter to the command:
+```
+pdk convert ?--template_url? my/template/location (the exact flag tbd)
```
Comment (DavidS) [inline README.md]: `--template-url` is explained at https://puppet.com/docs/pdk/1.0/pdk_reference.html

Comment (DavidS) [reply README.md]: (and this will use the same definition)

File: README.md [hunk 2]
```diff
@@ -102,6 +102,44 @@ pdk test unit
 
 This command runs all available unit tests.
 
+### Convert an existing module
+
+Aside from creating new modules, the PDK now has the ability to convert an existing module into a PDK ?enabled? module. You can then manage and test your module using the tools bundled with the PDK.
```
Comment (DavidS) [inline README.md]: Let's say "PDK **managed** module" for now. Makes it clear that the PDK has an active interest in the module's contents/setup. 

cc @rickmonro

File: README.md [hunk 3]
```diff
@@ -103,6 +103,42 @@ pdk test unit
...
+### Convert an existing module
+
+Aside from creating new modules, the PDK now has the ability to convert an existing module into a PDK managed module. You can then manage and test your module using the tools bundled with the PDK.
+
+To convert a module you must first decide on a template to convert to. This as default is set as ?PUPPET TEMPLATE?, however you can use your own template by passing it in as a parameter to the command:
+```
+pdk convert --template-url my/template/location
+```
+
+1. To start this process, in the existing module's directory, run:
```
Comment (bmjen) [inline README.md]: `To start this process with the default module template`

File: README.md [hunk 4]
```diff
@@ -103,6 +103,42 @@ pdk test unit
...
+
+```
+pdk convert
+```
+
+This command will then proceed to output a report based on converting your module to the new pdk template, informing you what changes are required.
```
Comment (bmjen) [inline README.md]: -`new`

File: README.md [hunk 5]
```diff
@@ -103,6 +103,42 @@ pdk test unit
...
+
+You will then be prompted with the choice to apply the changes or to abandon the process. Please note that although some of the changes can be automatically fixed, some may need manual intervention.
+
+Regardless of your choice, the report generated by the command will be put in ?FILEEEE? for review. (Please note this file will be overridden each time you run the command to reflect the current state.)
```
Comment (bmjen) [inline README.md]: s/`overridden`/`overwritten`

File: pdkconvertfailure.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+# pdk convert failure occurence
+
+As a user I am particularily brave and decided to run convert with the --force option. I ran this within my existing modules directory and the convert proceeded as follows.
```
Comment (bmjen) [inline pdkconvertfailure.md]: s/`particularily`/`particularly`

File: pdkconvertnoop.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# pdk convert noop
+
+As a user, I am simply intrigued by what it would take to convert my module to be managed by pdk. To this effect I have decided to go with a safe 'no-op' mode when it comes to running the pdk convert command.
```
Comment (bmjen) [inline pdkconvertnoop.md]: s/`managed`/`compatible`

Comment (HelenCampbell) [reply pdkconvertnoop.md]: What term we should use is still in dispute, I'll bring it up at grooming.

**Pull request closed without merging**